cmd/kubeless/function/deploy.go

```diff
@@ -177,11 +190,31 @@ var deployCmd = &cobra.Command{
 		}
 
 		f, err := getFunctionDescription(funcName, ns, handler, file, funcDeps, runtime, runtimeImage, mem, cpu, timeout, imagePullPolicy, port, headless, envs, labels, secrets, defaultFunctionSpec)
-
 		if err != nil {
 			logrus.Fatal(err)
 		}
 
+		if dryrun == true {
+			if output == "json" {
+				j, err := json.MarshalIndent(f, "", "    ")
+				if err != nil {
+					logrus.Fatal(err)
+				}
+				fmt.Println(string(j[:]))
+				return
+			}  else if output == "yaml" {
+				y, err := yaml.Marshal(f)
+				if err != nil {
+					logrus.Fatal(err)
+				}
+				fmt.Println(string(y[:]))
+				return
+			} else {
+				logrus.Infof("Output format needs to be yaml or json")
```

Comment:
`logrus.Fatalf` to return an error.

Reply:
yeah, I debated about that one actually. Because it is not really an error. but I will switch